source/swap.js

```diff
@@ -0,0 +1,40 @@
+import _curry3 from './internal/_curry3.js';
+
+/**
+ * Swap an item, at index `indexA` with another item, at index `indexB`, in a list of elements.
+ * A new list will be created containing the new elements order.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig Number -> Number -> [a] -> [a]
+ * @param {Number} indexA The first index
+ * @param {Number} indexB The second index
+ * @param {Array} list The list which will serve to realise the swap
+ * @return {Array} The new list reordered
+ * @example
+ *
+ *      R.swap(0, 2, ['a', 'b', 'c', 'd', 'e', 'f']); //=> ['c', 'b', 'a', 'd', 'e', 'f']
+ *      R.swap(-1, 0, ['a', 'b', 'c', 'd', 'e', 'f']); //=> ['f', 'b', 'c', 'd', 'e', 'a'] list rotation
+ */
+var swap = _curry3(function(indexA, indexB, list) {
+  var length = list.length;
+  var result = list.slice();
+  var positiveIndexA = indexA < 0 ? length + indexA : indexA;
+  var positiveIndexB = indexB < 0 ? length + indexB : indexB;
+  var positiveMin = Math.min(positiveIndexA, positiveIndexB);
+  var positiveMax = Math.max(positiveIndexA, positiveIndexB);
+
+  return positiveIndexA < 0 || positiveIndexA >= list.length
+      || positiveIndexB < 0 || positiveIndexB >= list.length
+      || positiveIndexA == positiveIndexB
+    ? list
```

Comment:
I think this should probably be a shallow clone of the list, perhaps `list.slice(0)`.

In all other cases here we return a new list.  Having a no-op case that returns the original value has caused us some issues in the past.  I'd like to avoid them.

Reply:
Could we leverage the existing `result` local instead of re-slicing? Also is there a benefit to calling `slice(0)` vs. `slice()`?

Reply:
> Could we leverage the existing `result` local instead of re-slicing?

Yes, of course.  Anything that achieves that idea is fine.

> Also is there a benefit to calling `slice(0)` vs. `slice()`?

No, probably not.  `slice(0)` has always been my habit, but not for any good reason.

Reply:
Thanks for the feedback! I have rebased, and updated the function to always return a new result.